lib/DataQueue.js

```diff
@@ -0,0 +1,207 @@
+    }
+
+    this.name = name;
+    this.library = library ? library : '*CURLIB';
+    this.qualifiedName = this.name.padEnd(10, ' ') + this.library.padEnd(10, ' ');
+    this.maxLength = undefined;
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Create Data Queue (CLDTAQ) CL Command
+  //////////////////////////////////////////////////////////////////////////////
+  /**
+   * @description Creates a Data Queue on the system of the connection passed in
+   *              the constructor.
+   * @constructor
+   * @param {number} maxLength
+   * @param {function} callback
+   */
+  create(maxLength, callback) {
+    console.log(CommandCall.CommandCall);
+    this.maxLength = maxLength;
+    const commandConfig = {
+      type: 'cl',
+      command: `CRTDTAQ DTAQ(${this.library ? `${this.library}/` : ''}${this.name}) MAXLEN(${this.maxLength})`
+    }
+
+    this.connection.add(new CommandCall(commandConfig));
+    this.connection.run((error, xmlOutput) => {
+      if (error) {
+        return callback(error);
+      }
+
+      const result = xmlToJson(xmlOutput);
+      callback(null, result);
+    });
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Delete Data Queue (DLTDTAQ) CL Command
+  //////////////////////////////////////////////////////////////////////////////
+  delete(callback) {
+    const commandConfig = {
+      type: 'cl',
+      command: `DLTDTAQ DTAQ(${this.library ? `${this.library}/`: ''}${this.name})`
+    }
+
+    this.connection.add(new CommandCall(commandConfig));
+    this.connection.run((error, xmlOutput) => {
+      if (error) {
+        return callback(error);
+      }
+
+      const result = xmlToJson(xmlOutput);
+      callback(null, result);
+    });
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Receive Data Queue (QRCVDTAQ) API
+  //////////////////////////////////////////////////////////////////////////////
+
+  async receive(wait, length, callback) {
+    if (typeof wait === 'function') {
+      callback = wait;
+      wait = 0;
+    }
+
+    if (length == undefined) {
+      if (this.maxLength == undefined) {
+        const description = await this._getDataQueueDescription();
+        this.maxLength = description[0].data[2].value;
+      }
+
+      length = this.maxLength;
+    }
+
+    const program = new ProgramCall('QRCVDTAQ', { lib: 'QSYS' });
+
+    program.addParam(this.name,    QRCVDTAQParameters.DataQueueName.type, { io: QRCVDTAQParameters.DataQueueName.io });
+    program.addParam(this.library, QRCVDTAQParameters.LibraryName.type,   { io: QRCVDTAQParameters.LibraryName.io });
+    program.addParam(length,       QRCVDTAQParameters.LengthOfData.type,  { io: QRCVDTAQParameters.LengthOfData.io });
+    program.addParam('',           `${this.maxLength}A`,                  { io: QRCVDTAQParameters.Data.io });
```

Comment:
We talked about how to do this for "structures" (the array) to get the `len` and then using `setlen`, but can I do the same for single values? And should I here, with the `LengthOfData` parameter?